scripts/generator/graphql_generator/csharp/SDK/CheckoutMutator.cs.erb

```diff
@@ -0,0 +1,268 @@
+                if (error != null) {
+                    callback(error);
+                    return;
+                }
+
+                HandleCheckoutMutation(
+                    response.checkoutCreate().checkout(),
+                    response.checkoutCreate().userErrors(),
+                    CheckoutPoller.PollCheckoutReady,
+                    callback
+                );
+            });
+        }
+
+        public void UpdateItems(
+            Checkout checkout, 
+            List<CheckoutLineItemInput> lineItemsToAdd, 
+            List<string> lineItemsToRemove,
+            CompletionCallback callback
+        ) {
+            MutationQuery query = new MutationQuery();
+            DefaultQueries.checkout.LineItemsRemove(query, checkout.id(), lineItemsToRemove);
+            DefaultQueries.checkout.LineItemsAdd(query, checkout.id(), lineItemsToAdd);
+
+            Client.Mutation(query, (Mutation response, ShopifyError error) => {
+                if (error != null) {
+                    callback(error);
+                    return;
+                }
+
+                HandleCheckoutMutation(
+                    response.checkoutLineItemsAdd().checkout(),
+                    response.checkoutLineItemsAdd().userErrors(),
+                    CheckoutPoller.PollCheckoutReady,
+                    callback
+                );
+            });
+        }
+
+        public void SetShippingLine(
+            Checkout checkout,
+            string shippingRateHandle, 
+            CompletionCallback callback
+        ) {
+            MutationQuery query = new MutationQuery();
+            DefaultQueries.checkout.ShippingLineUpdate(query, checkout.id(), shippingRateHandle);
+
+            Client.Mutation(query, (Mutation response, ShopifyError error) => {
+                if (error != null) {
+                    callback(error);
+                    return;
+                }
+
+                HandleCheckoutMutation(
+                    response.checkoutShippingLineUpdate().checkout(),
+                    response.checkoutShippingLineUpdate().userErrors(),
+                    CheckoutPoller.PollCheckoutReady,
+                    callback 
+                );
+            });
+        }
+
+        public void SetEmailAddress(Checkout checkout, string email, CompletionCallback callback) {
+            MutationQuery query = new MutationQuery();
+            DefaultQueries.checkout.EmailUpdate(query, checkout.id(), email);
+
+            Client.Mutation(query, (Mutation response, ShopifyError error) => {
+                if (error != null) {
+                    callback(error);
+                    return;
+                }
+
+                HandleCheckoutMutation(
+                    response.checkoutEmailUpdate().checkout(),
+                    response.checkoutEmailUpdate().userErrors(),
+                    CheckoutPoller.PollCheckoutAvailableShippingRatesReady,
```

Comment:
I just realized this line should be `CheckoutPoller.PollCheckoutReady` since Email doesn't affect available shipping rates